src/usng.ts

```diff
@@ -0,0 +1,1397 @@
+// Copyright (c) 2009 Larry Moore, larmoor@gmail.com
+//               2014 Mike Adair, Richard Greenwood, Didier Richard, Stephen Irons, Olivier Terral and Calvin Metcalf (proj4js)
+//               2014 Codice Foundation
+// Released under the MIT License; see
+// http://www.opensource.org/licenses/mit-license.php
+// or http://en.wikipedia.org/wiki/MIT_License
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+/*global define,Math,parseInt,parseFloat*/
+
+type LL = {
```

Comment:
❗️ Introducing TypeScript is going well beyond the scope of change needed for this addition. Additionally, this library will hopefully be superseded by a cross-platform alternative within the next year in which case it will be deprecated so future-proofing this version seems to be a poor waste of time and effort as it simultaneously introduces greater risk to this version.

Reply:
To future readers: see [this comment](https://github.com/codice/usng.js/pull/18#issuecomment-424845915) for the resolution.